src/VisualStudio/Core/SolutionExplorerShim/DiagnosticItem/DiagnosticItem.cs

```diff
@@ -154,15 +148,34 @@ private void UpdateRuleSetFile(string pathToRuleSet, ReportDiagnostic value)
             var analyzerID = _analyzerItem.AnalyzerReference.Display;
             var rules = FindOrCreateRulesElement(ruleSetDocument, analyzerID);
             var rule = FindOrCreateRuleElement(rules, _descriptor.Id);
-            rule.Attribute("Action").Value = newAction;
+
+            if (value == ReportDiagnostic.Default)
+            {
+                // If the new severity is 'Default' we just delete the entry for the rule from the ruleset file.
+                // In the absence of an explicit entry in the ruleset file, the rule reverts back to its 'Default' severity.
```

Comment:
No, you don't need to handle included ruleset case. If you want to edit those, you need to use the rule set editor.